From faa9fe0d844cf9bd0cf7fb13c2d79b7583487c1f Mon Sep 17 00:00:00 2001
From: Petr Oros <poros@redhat.com>
Date: Thu, 23 Oct 2014 12:22:25 -0400
Subject: [redhat] kabi: show-kabi with -m respecting -r parameter

Message-id: <6fc49a14b1d7235fdc48852c330d7e7e1913b0cb.1414066546.git.poros@redhat.com>
Patchwork-id: 98312
O-Subject: [RHEL7.1 PATCHv2 2/2] kabi: show-kabi with -m respecting -r parameter
Bugzilla:
RH-Acked-by: Jarod Wilson <jarod@redhat.com>

when use show-kabi with -m param ( generate Module.kabi_ARCH file )
        -r param is ignored. It's bad, because removed symbols is also added
        into output and kabi-check will fail on removed symbols.

Signed-off-by: Petr Oros <poros@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/redhat/kabi/show-kabi b/redhat/kabi/show-kabi
index 55d603f..f864d98 100755
--- a/redhat/kabi/show-kabi
+++ b/redhat/kabi/show-kabi
@@ -82,11 +82,6 @@ def show_kabi(kabi, order, release, fmt, show_head):
             if kabi[current_arch][sym][0][0] != "#":
                 print "Invalid metadata format: " + kabi[current_arch][sym][0]
                 sys.exit(1)
-            # format Module.kabi_$arch styled file
-            # note: keep checksum for removed symbols
-            if fmt == "module":
-                print kabi[current_arch][sym][1]
-                continue
             minor_range = kabi[current_arch][sym][0][1:-1].split("-", 2)
             minor_range[0] = int(minor_range[0])
             if minor_range[1] == "":
@@ -97,8 +92,12 @@ def show_kabi(kabi, order, release, fmt, show_head):
                 continue
             if (minor_range[1] <= rhel_minor) and (minor_range[1] != sys.maxint):
                 continue
+            # format Module.kabi_$arch styled file
+            if fmt == "module":
+                print kabi[current_arch][sym][1]
             # format kabi_whitelist_$arch styled file
-            print "\t" + sym
+            else:
+                print "\t" + sym
 
 def usage():
     print """show-kabi:
-- 
1.7.1